loader.js

```diff
@@ -8,9 +8,13 @@ var NodeTargetPlugin = require("webpack/lib/node/NodeTargetPlugin");
 var LibraryTemplatePlugin = require("webpack/lib/LibraryTemplatePlugin");
 var SingleEntryPlugin = require("webpack/lib/SingleEntryPlugin");
 var LimitChunkCountPlugin = require("webpack/lib/optimize/LimitChunkCountPlugin");
+
+var extractTextPluginHmrRuntime = require.resolve("./hotModuleReplacement.js");
+
 module.exports = function(source) {
 	if(this.cacheable) this.cacheable();
-	return source;
+	// Even though this gets overwritten if extract+remove are true, without it, the runtime doesn't get added to the chunk
+	return "if(module.hot){require('" + extractTextPluginHmrRuntime + "');}\n" + source;
```

Comment:
I was originally hoping to get this into 1.x (this PR is off of the `webpack-1` branch), which I'm assuming has to support Node <4. If that's out of the question, I suppose I can run off of my fork until I eventually get onto webpack 2.